cmd/entrypoint/waiter_test.go

```diff
@@ -122,3 +123,58 @@ func TestRealWaiterWaitWithContent(t *testing.T) {
 		t.Errorf("expected Wait() to have detected a non-zero file size by now")
 	}
 }
+
+func TestRealWaiterWaitWithErrorWaitfile(t *testing.T) {
+	tmp, err := ioutil.TempFile("", "real_waiter_test_file*.err")
+	if err != nil {
+		t.Errorf("error creating temp file: %v", err)
+	}
+	tmpFileName := strings.Replace(tmp.Name(), ".err", "", 1)
+	defer os.Remove(tmp.Name())
+	rw := realWaiter{}
+	doneCh := make(chan struct{})
+	go func() {
+		// error of type skipError is returned after encountering a error waitfile
+		err := rw.setWaitPollingInterval(testWaitPollingInterval).Wait(tmpFileName, false, false)
+		if err == nil {
+			t.Errorf("expected skipError upon encounter error waitfile")
+		}
+		switch typ := err.(type) {
+		case skipError:
+			close(doneCh)
+		default:
+			t.Errorf("unexpected error type %T", typ)
+		}
+	}()
+	select {
+	case <-doneCh:
+		// Success
+	case <-time.After(2 * testWaitPollingInterval):
+		t.Errorf("expected Wait() to have detected a non-zero file size by now")
+	}
+}
+
+func TestRealWaiterWaitWithBreakpointOnFailure(t *testing.T) {
+	tmp, err := ioutil.TempFile("", "real_waiter_test_file*.err")
+	tmpFileName := strings.Replace(tmp.Name(), ".err", "", 1)
```

Comment:
If `TempFile` returns an error, `tmp` will be nil, and this will panic. Can we move this below the `err != nil` check on the next line?